### PR TITLE
Zone activity - farm plot planting

### PR DESCRIPTION
**Pull request by Robik81**
#### Summary

```SUMMARY: Features "adds new option for Farm: Plot zone to attach specific seed and new activity to plant in nearby plot zones"```

#### Purpose of change

As outlined in #24826 , this PR aims at reducing tedium of farming. Follow up of #25125 . Planting activity for Farm: Plot zone. Watering, weeding and fertilizing will follow in separate PRs.

#### Describe the solution

1. Select seed for existing Farm: Plot zone (via Edit->Edit options in zone manager) or select seed while creating a new one
2. After Tilling farm plots activity, perform Plant seeds activity (shortcut Shift+O).
3. Watch your character do all the work automatically.

#### Update
Watering, weeding and fertilizing will follow if / when the improved crop farming is implemented again.

### CODE REVIEW

**Comment by Vasyan2006** [comment]
I'm still planning to separate >Watering, weeding and fertilizing into new functions, so you will be able to use your tumba-jumba zone magic and apply them over every tile. I suppose it will be faster to commit such changes into your branch rather then opening new PR.

**Comment by Robik81** [comment]
> I suppose it will be faster to commit such changes into your branch rather then opening new PR.

I think if you'l make tiny PR only with refactoring of a function into two, @ZhilkinSerg can review and merge something like this quite fast?

File: src/iexamine.cpp [hunk 1]
```diff
@@ -1663,16 +1638,28 @@ void iexamine::dirtmound( player &p, const tripoint &examp )
...
+ */
+int iexamine::query_seed( const player &p, const std::vector<seed_tuple> &seed_entries )
+{
     // Don't use y/n prompt, stick with one kind of menu
     uimenu smenu;
-    smenu.return_invalid = true;
+
+    // if true, it works fine for normal planting, but closes immediately if called
+    // from parent uimenu when adding new planting zone
+    smenu.return_invalid = false;
```
[the remark this answers is not in the document]
Comment (Robik81) [reply src/iexamine.cpp]: Your guess was correct. That was indeed a problem. 

Thank you very much for helping me out.

(I see you are going to handle it on the uimenu side, cool.)

**Comment by ZhilkinSerg** [comment]
#25204 is merged

**Comment by Robik81** [comment]
Thank you guys.

On the weeding I go... it will be in separate PR though, this one is big enough as it is.

**Comment by Robik81** [comment]
Given that PR #24291 was rolled back, I updated this PR to work with the old planting system.

File: src/iexamine.cpp [hunk 2]
```diff
@@ -1613,6 +1612,80 @@ void iexamine::fungus(player &p, const tripoint &examp)
...
+    if( item::count_by_charges( seed_id ) ) {
+        used_seed = p.use_charges( seed_id, 1 );
+    } else {
+        used_seed = p.use_amount( seed_id, 1 );
+    }
+    used_seed.front().set_age( 0 );
+    g->m.add_item_or_charges( examp, used_seed.front() );
+    g->m.set( examp, t_dirt, f_plant_seed );
+    p.moves -= 500;
+    add_msg( _( "Planted %s" ), item::nname( seed_id ).c_str() );
```
Comment (ZhilkinSerg) [inline src/iexamine.cpp]: Should probably add period after `Planted %s`

**Comment by ZhilkinSerg** [comment]
No feedback for player (in message log or popup) when `Plant items` action is selected, but plant zone is not defined or plant zone has no seeds selected in its options.

**Comment by Robik81** [comment]
> No feedback for player (in message log or popup) when Plant items action is selected, but plant zone is not defined or plant zone has no seeds selected in its options.

There is feedback if no compatible zone is nearby:
image

There is no feedback if you have only Unsorted loot zone defined but there is nothing to sort (i'l write it down as sort loot issue). You can't pick any farm related action in this case though.

If you have plant zone, but is not tilled, have no seed attached, or you have no seeds, there is also not any feedback. This one is probably the one you are referring to.

I'll fix the issue.

**Comment by Robik81** [comment]
#### New commits changes

- extra dot in Planted %s message
- action menu now checks if you have any seeds at all (planting disabled if not)
- added message "You planted all seeds you could." after planting action is finished (whether you planted anything or not)

**Pull request merged**